File: code/java/src/main/java/abstractfactory/FantasyLandFactory.java
package abstractfactory;

class FantasyLandFactory implements AbstractFoodStallFactory {
    @Override
    public IceCream createIceCream() {

        System.out.println("Creating magical-themed ice cream");
        return new MagicalIceCream();
    }

    @Override
    public Pizza createPizza() {
        System.out.println("Creating enchanted-themed pizza");
        return new EnchantedPizza();
    }
}

File: code/typescript/src/abstractFactory/FantasyLandFactory.ts
import { AbstractFoodStallFactory } from './AbstractFoodStallFactory';
import { IceCream } from './IceCream';
import { Pizza } from './Pizza';
import { MagicalIceCream } from './MagicalIceCream';
import { EnchantedPizza } from './EnchantedPizza';

export class FantasyLandFactory implements AbstractFoodStallFactory {
    createIceCream(): IceCream {
        console.log("Creating magical-themed ice cream");
        return new MagicalIceCream();
    }

    createPizza(): Pizza {
        console.log("Creating enchanted-themed pizza");
        return new EnchantedPizza();
    }
}
